### PR TITLE
Devnet faucet in DW

### PR DESCRIPTION
**Pull request by mvaivre**

### CODE REVIEW

File: apps/desktop-wallet/src/storage/global/globalActions.ts [hunk 1]
```diff
@@ -60,7 +62,46 @@ export const appDataClearFailed = createAction('app/appDataClearFailed')
 
 export const receiveFaucetTokens = createAsyncThunk<SentTransaction, AddressHash, { rejectValue: SnackbarMessage }>(
   'assets/receiveFaucetTokens',
-  async (destinationAddress: AddressHash, { rejectWithValue, fulfillWithValue }) => {
+  async (destinationAddress: AddressHash, { getState, rejectWithValue, fulfillWithValue }) => {
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/storage/global/globalActions.ts:65]: Yes absolutely, I'm checking for devnet but I'll add a check for testnet too.

[the remark this answers is not in the document]
Comment (mvaivre) [reply apps/desktop-wallet/src/storage/global/globalActions.ts:65]: Done

[the remark this answers is not in the document]
Comment (polarker) [reply apps/desktop-wallet/src/storage/global/globalActions.ts:65]: great!

File: apps/desktop-wallet/src/storage/global/globalActions.ts [hunk 2]
```diff
@@ -58,9 +60,54 @@ export const appDataCleared = createAction('app/appDataCleared')
...
+      timestamp: new Date().getTime(),
+      status: 'sent',
+      type: 'faucet'
+    }
+
+    if (currentNetwork.name === 'devnet') {
+      throttledClient.init(currentNetwork.settings.nodeHost, currentNetwork.settings.explorerApiHost)
+
+      const faucetPublicKey = '0381818e63bd9e35a5489b52a430accefc608fd60aa2c7c0d1b393b5239aedf6b0'
+      const faucetPrivateKey = 'a642942e67258589cd2b1822c631506632db5a12aabcf413604e785300d762a5'
```
Comment (nop33) [inline apps/desktop-wallet/src/storage/global/globalActions.ts:88]: ahm.... @mvaivre are we should this is safe? Can't someone just drain the testnet faucet now?

**Comment by nop33** [comment]
Related issue:
- #954

**Pull request merged**